catalog/rest/rest.go

```diff
@@ -375,10 +376,7 @@ func handleNon200(rsp *http.Response, override map[int]error) error {
 	return e
 }
 
-func fromProps(props iceberg.Properties) *options {
-	o := &options{
-		additionalProps: iceberg.Properties{},
-	}
+func fromProps(props iceberg.Properties, o *options) {
```

Comment:
I changed the signature to make it less error-prone to round-trip the options into properties and back. This is done when creating a client, to convert options to properties, then merge in any properties retrieved from the server, and then convert back to options.

This way, we can easily clone the original options object and then override values from the properties. Previously, any time a new field was added to options that was not represented in properties (like the new `awsConfigSet` flag I added), if one forgot to update the list of manually copied fields ([here](https://github.com/apache/iceberg-go/pull/384/files#diff-b4f7aa1f4862334f1387eca97a977f5bcc7ba52a527562ad2f89cc4c767d618cL631-L632)), it would be easy to introduce a subtle bug.